virttest/libvirt_xml/vm_xml.py

```diff
@@ -772,6 +773,18 @@ def set_vm_vcpus(vm_name, vcpus, current=None, sockets=None, cores=None,
                                          'cores': cores,
                                          'threads': threads}
                 vmxml['cpu'] = vmcpu_xml
+            if update_numa and vmxml.cpu.numa_cell:
+                vmcpu_xml = vmxml['cpu']
+                if len(vmxml.cpu.numa_cell) == 2 and vcpus >= 4:
+                    node0xml = vmcpu_xml.numa_cell[0]
+                    node0xml["cpus"] = "0-%s" % ((vcpus / 2) - 1)
+                    node1xml = vmcpu_xml.numa_cell[1]
+                    node1xml["cpus"] = "%s-%s" % (vcpus / 2, vcpus - 1)
+                    vmcpu_xml.set_numa_cell([node0xml, node1xml])
```

Comment:
we can make it more generic,
```python
total_numa = len(vmxml.cpu.numa_cell)
if vcpus < total_numa:
    vcpus_numa = total_numa // vcpus
    vcpus_rem = total_numa % vcpus
    index = 0
    for each_numa in range(total_numa):
        node = {}
        if vcpus_numa > 1:
            if each_numa == total_numa - 1:
                node["cpus"] = "%s-%s" % (index, index + vcpus_numa + vcpus_rem - 1)
            else:
                node["cpus"] = "%s-%s" % (index, index + vcpus_numa - 1)
        else:
            node["cpus"] = str(vcpus_numa)
        numa_list.append(node)
    vmcpu_xml.set_numa_cell(numa_list)

else:
    logging.warning("Guest numa could not be updated, expect "
                    "failures if guest numa is checked")
```

Reply:
@balamuruhans make sense, i just wanted to rework on it based on bugzilla update, nevertheless we can implement the generic approach, will check on this and update it.

Reply:
as the intention of this change is to have a numa cell matched with no. of vcpus lets try to change in generic way as you have suggested if not possible lets fall back to having two nodes with vcpus divided across,

Reply:
we will have total_numa < vcpus and most of the times that is going to be the case.....